File: 451-sort-characters-by-frequency/sort-characters-by-frequency.py
class Solution:
    def frequencySort(self, s: str) -> str:
        res = []
        count = Counter(s)
        sorted_count = sorted(count.items(), key=lambda x: x[1], reverse=True)
        for c, freq in sorted_count:
            res.append(c * freq)
        return "".join(res)



        

File: 451-sort-characters-by-frequency/sort-characters-by-frequency.cpp
class Solution {
public:
    string frequencySort(string s) {
        unordered_map<char, int> mp;
        multimap<int, char> r;
        for (char c : s) {
            mp[c]++;
        }

        vector<pair<int, char>> freq;
        for (auto [k, v] : mp) {
            freq.push_back({v, k});
        }

        sort(freq.begin(), freq.end());
        reverse(freq.begin(), freq.end());

        string res = "";
        for (auto pair : freq) {
            res += string(pair.first, pair.second);
        }
        return res;
    }
};